src/OrbitBase/include/OrbitBase/TemporaryFile.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_TEMPORARY_FILE_H_
+#define ORBIT_BASE_TEMPORARY_FILE_H_
+
+#include "OrbitBase/File.h"
+
+namespace orbit_base {
+class TemporaryFile final {
+ public:
+  TemporaryFile() { Create(); }
+  ~TemporaryFile() { CloseAndRemove(); }
+
+  TemporaryFile(const TemporaryFile&) = delete;
+  TemporaryFile& operator=(const TemporaryFile&) = delete;
+  TemporaryFile(TemporaryFile&&) = delete;
+  TemporaryFile& operator=(TemporaryFile&&) = delete;
```

Comment:
Why shouldn't this class be moveable?

Reply:
Defining the move constructor but deleting the move assignment is super weird, they should both be implemented.